### PR TITLE
Update Ruby to 3.1

### PR DESCRIPTION
**Pull request by tpendragon**
Work towards #329 

Still needs the Ansible work.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -52,6 +52,13 @@ gem 'sneakers'
 gem 'sprockets', '>= 3.7.2'
 gem 'turbolinks'
 
+## Added for Ruby 3.1 support
+gem "matrix"
+gem "net-imap", require: false
+gem "net-pop", require: false
+gem "net-smtp", require: false
```
Comment (hackartisan) [inline Gemfile:59]: Question: did you find you needed all of these, or did you pull these from the work on another project? I saw them but the only one I needed to get tests passing for dpul was net-smtp so I didn't add the others. I'm wondering if I should.

**Review by hackartisan — APPROVED** [review]
This looks good, my question is more about https://github.com/pulibrary/dpul/pull/1414 lol so I'm not blocking.

File: Gemfile (hunk 1 above)
Comment (tpendragon) [reply Gemfile:59]: Oh I just copied them. Hm, I can try removing them - we don't send email..

**Comment by tpendragon** [comment]
Good call on the gems, I also only needed net-smtp.

**Pull request merged**